File: guides/micronaut-scheduled/java/src/main/java/example/micronaut/EmailTask.java
package example.micronaut;

public class EmailTask implements Runnable {

    private String email;
    private String message;
    private EmailUseCase emailUseCase;

    public EmailTask(EmailUseCase emailUseCase, String email, String message) {
        this.email = email;
        this.message = message;
        this.emailUseCase = emailUseCase;
    }

    @Override
    public void run() {
        emailUseCase.send(email, message);
    }

}

File: guides/micronaut-scheduled/python/src/example/micronaut/email_task.py
from java.lang import Runnable

from .email_use_case import EmailUseCase


class EmailTask(Runnable):
    def __init__(self, email_use_case: EmailUseCase, email: str, message: str):
        self.email_use_case = email_use_case
        self.email = email
        self.message = message

    def run(self) -> None:
        self.email_use_case.send(self.email, self.message)
